pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskExecutor.java

```diff
@@ -77,4 +77,14 @@ protected SegmentZKMetadataCustomMapModifier getSegmentZKMetadataCustomMapModifi
         .singletonMap(MinionConstants.PurgeTask.TASK_TYPE + MinionConstants.TASK_TIME_SUFFIX,
             String.valueOf(System.currentTimeMillis())));
   }
+
+  private static SegmentPurger.RecordPurger getRecordPurger(PinotTaskConfig pinotTaskConfig, String rawTableName,
+      SegmentPurger.RecordPurgerFactory recordPurgerFactory, TableConfig tableConfig, Schema schema) {
+    if (recordPurgerFactory == null) {
+      return null;
+    }
+    SegmentPurger.RecordPurger recordPurger =
+        recordPurgerFactory.getRecordPurger(rawTableName, pinotTaskConfig, tableConfig, schema);
+    return recordPurger != null ? recordPurger : recordPurgerFactory.getRecordPurger(rawTableName);
```

Comment:
Is it better to just call `getRecordPurger(rawTableName)` in `getRecordPurger(rawTableName, pinotTaskConfig, tableConfig, schema)`?

Reply:
Modified